src/Mvc/Mvc.RazorPages/src/Infrastructure/PageActionEndpointDataSource.cs

```diff
@@ -40,15 +40,15 @@ internal sealed class PageActionEndpointDataSource : ActionEndpointDataSourceBas
     // selection. Set to true by builder methods that do dynamic/fallback selection.
     public bool CreateInertEndpoints { get; set; }
 
-    protected override List<Endpoint> CreateEndpoints(IReadOnlyList<ActionDescriptor> actions, IReadOnlyList<Action<EndpointBuilder>> conventions)
+    protected override List<Endpoint> CreateEndpoints(IReadOnlyList<ActionDescriptor> actions, IReadOnlyList<Action<EndpointBuilder>> conventions, IReadOnlyList<Action<EndpointBuilder>> finalConventions)
     {
         var endpoints = new List<Endpoint>();
         var routeNames = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
         for (var i = 0; i < actions.Count; i++)
         {
             if (actions[i] is PageActionDescriptor action)
             {
-                _endpointFactory.AddEndpoints(endpoints, routeNames, action, Array.Empty<ConventionalRouteEntry>(), conventions, CreateInertEndpoints);
+                _endpointFactory.AddEndpoints(endpoints, routeNames, action, Array.Empty<ConventionalRouteEntry>(), conventions, finalConventions, CreateInertEndpoints);
```

Comment:
I guess I need to finish reviewing #42557 so this PR can react to the better group support.

Reply:
Yep, now that it's been merged I'll have to do quite a bit of rebasing on this PR. 😅